crates/core/tedge_api/src/builder.rs

```diff
@@ -21,9 +22,14 @@ impl ThinEdgeJsonBuilder {
             return Err(ThinEdgeJsonBuilderError::EmptyThinEdgeJsonRoot);
         }
 
+        if self.measurement_type.is_none() {
+            self.measurement_type = Some("ThinEdgeMeasurement".to_string());
```

Comment:
Why set a value here when none is provided?

Previously the default was set by the [c8y mapper](https://github.com/thin-edge/thin-edge.io/pull/1911/files#diff-0f2505e467ef345eedca89e2de8d1699847df2df3fef95767766d23a5f5e76b5L45-L46) and this must still be the case as this is *specific* to Cumulocity.

Reply:
I had to remove this because if the default is set as before then, it's really hard to update the default value with the `given type & value`, as it's stored in a vector of `u8`. That is the reason why I took this direction.

Reply:
What I'm saying is that, if unset in the incoming message, the `measurement_type` must be kept as `None` and the c8y default value should be pushed by the c8y mapper only in none has been pushed during the message translation.